### PR TITLE
fix the device crach appeared when the device fallback is performed

### PR DESCRIPTION
**Pull request by yiwenxue**
Re: #https://github.com/cocos/3d-tasks/issues/12474#issue-1255571596

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by dumganhar — APPROVED** [review]

File: native/cocos/renderer/gfx-agent/DeviceAgent.cpp [hunk 1]
```diff
@@ -92,14 +93,20 @@ bool DeviceAgent::doInit(const DeviceInfo &info) {
...
-        _mainMessageQueue, DeviceDestroy,
-        actor, _actor,
-        {
-            if (actor) {
-                actor->destroy();
-            }
-        });
+    if (!_mainMessageQueue) {
+        if(_actor) {
+            _actor->destroy();
```
Comment (dumganhar) [inline native/cocos/renderer/gfx-agent/DeviceAgent.cpp]: Does it need to set _actor to nullptr?

Comment (yiwenxue) [reply native/cocos/renderer/gfx-agent/DeviceAgent.cpp]: No need. The actor is intrusive and can be deleted automatically.

**Comment by suqionglian** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/gfx-agent/DeviceAgent.cpp (hunk 1 above)
Comment (minggo) [reply native/cocos/renderer/gfx-agent/DeviceAgent.cpp]: It is not IntrusivePtr now. It is a raw pointer. It is deleted in DeviceAgent destructor.

**Review by star-e — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**